base/tuple.jl

```diff
@@ -565,6 +565,20 @@ function isless(t1::Any32, t2::Any32)
     return n1 < n2
 end
 
+# Performance optimization to pack pairs of integers into a single integer
+# for comparison. This is useful for sorting tuples of integers.
+# TODO: remove this when the compiler can optimize the generic version better
+# See #48724 and #48753
+_pack_tuple_promote(a::Unsigned,b) = unsigned(promote(a,b)[1])
+_pack_tuple_promote(a::Signed,b) = signed(promote(a,b)[1])
+_pack_tuple((a,b)) = widen(_pack_tuple_promote(a,b)) << 8sizeof(b) - typemin(b) + b
+let T = Union{Base.BitIntegerSmall, Int, UInt}
+    function isless(a::Tuple{T, T}, b::Tuple{T, T})
+        typeof(a) == typeof(b) || return invoke(isless, NTuple{2, NTuple{2}}, a, b)
+        isless(_pack_tuple(a), _pack_tuple(b))
+    end
+end
```

Comment:
```suggestion
_pack_tuple((a,b)) = widen((a)) << 8sizeof(b) - typemin(b) + b
for T in unique((BitUnsignedSmall_types..., UInt))
    Ts = signed(T)
    @eval isless(a::Tuple{T1, T2}, b::Tuple{T1, T2}) where {T1<:Union{$T,$Ts}, T2<:Union{$T,$Ts}} = isless(_pack_tuple(a), _pack_tuple(b))
end
```

How about this, which only redefines isless for tuples with elements of the same size?

Reply:
I thought that `isless((1, 1), (1, unsigned(2)))` would fail there because I misunderstood dispatch. I think tuples dispatch differently than other parametric types so this works. Thanks!